sdk/core/core/src/az_json_builder.c

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <az_json_builder.h>
+
+#include <az_str.h>
+
+#include <_az_cfg.h>
+
+AZ_NODISCARD az_result
+az_json_builder_init(az_json_builder * const out, az_span_action const write) {
```

Comment:
I don't understand that use/value of az_span_action. What is the purpose of this?

Reply:
Also, I think *_init takes a mutable span as an argument and associates it with the az_json_builder.